backend/module/chaos/serverexperimentation/xds/xds_integration_test.go

```diff
@@ -78,18 +82,32 @@ func TestEnvoyFaultsTimeBasedTermination(t *testing.T) {
 
 	criteria := &testCriteria{}
 
-	terminator := terminator.NewTestMonitor(
-		ts.Storer,
-		[]string{"type.googleapis.com/clutch.chaos.serverexperimentation.v1.HTTPFaultConfig"},
-		[]terminator.TerminationCriteria{criteria},
-		ts.Logger.Sugar(),
-		ts.Scope)
+	terminator.CriteriaFactories["type.googleapis.com/google.protobuf.StringValue"] = &testCriteriaFactory{testCriteria: criteria}
```

Comment:
Now that I look at this - should this test be in `xds_integration_test` file?

Reply:
Up for discussion I guess: it verifies that we actually stop experiments at the xDS level during a termination, but if you think it makes sense to move to another file I can do that.

There are potentially some work that should be done around the integration test framework to better support multiple files, but I can take a look at it

Reply:
let's keep it the way it is for now then